portfolio/src/main/webapp/script.js

```diff
@@ -46,7 +46,35 @@ function rotateImages() {
   setTimeout("rotateImages()", time);
 }
 
+function showHideCommentForm() {
+  fetch('/login-status')
+    .then(response => response.json())
+    .then((json) => {
+      if (json['loggedIn'].localeCompare("true")) {
```

Comment:
I would rephrase this conditional more straightforwardly as:

```
if (json['loggedIn'] === 'true') {
  // do logged-in things
} else {
  // do logged-out things
}
```

`localeCompare()` compares the lexical order of two strings, and while this code looks to be _technically_ correct, it is in my opinion somewhat misleading and confusing.